MC/bin/o2dpg_sim_workflow_anchored.py

```diff
@@ -295,12 +296,52 @@ def determine_timestamp(sor, eor, splitinfo, cycle, ntf, HBF_per_timeframe = 256
     production_offset = int(thisjobID * maxcycles) + cycle
     # add the time difference of this slot to start-of-run to get the final timestamp
     timestamp_of_production = sor + production_offset * ntf * HBF_per_timeframe * LHCOrbitMUS / 1000
-    # this is a closure test. If we had prefect floating point precision everywhere, it wouldn't fail.
+    # this is a closure test. If we had perfect floating point precision everywhere, it wouldn't fail.
     # But since we don't have that and there are some int casts as well, better check again.
     assert (timestamp_of_production >= sor)
     assert (timestamp_of_production <= eor)
     return int(timestamp_of_production), production_offset
 
+
+def exclude_timestamp(ts, orbit, run, filename):
+    """
+    Checks if timestamp ts (or orbit) falls within a bad data period.
+    Returns true if this timestamp should be excluded; false otherwise
+    
+    ts is supposed to be in milliseconds
+    orbit is some orbit after the orbitreset of the run
+    """
+    if len(filename) == 0:
+       return False
+
+    if not os.path.isfile(filename):
+       return False
+
+    # read txt file into a pandas dataframe ---> if this fails catch exception and return
+    df = pd.read_csv(filename, header=None, names=["Run", "From", "To", "Message"])
```

Comment:
Note that the CTFReader accepts delimiters like `,`, `;`, space and tab.

Reply:
I did this based on the example file that was given to me. Wouldn't it be easier to have a simple rule for this file? 
And I am not sure that space as a delimiter is useful because the example file contained words like `FC unstable` which would probably be split incorrectly.

Will make the code more generic tomorrow. For now it works and I need to have it in the nightly tag for testing tomorrow/